### PR TITLE
Hide result variables based on role

### LINKED ISSUE #1998: Explorer & Hierarchy: Result variables should not be shown for Data Cleanser and Data Editor

### PR DESCRIPTION
**Pull request by ramirobg94**
resolves #1998

### CODE REVIEW

File: webapp/store/user/hooks.js [hunk 1]
```diff
@@ -25,6 +25,9 @@ export const useAuthCanViewTemplates = () => Authorizer.canViewTemplates(useUser
 export const useAuthCanCreateTemplate = () => Authorizer.canCreateTemplate(useUser())
 export const useAuthCanEditTemplates = () => Authorizer.canEditTemplates(useUser())
 
+// ====== Auth / Analysis
+export const useAuthcanUseAnalysis = () => Authorizer.canAnalyzeRecords(useUser(), useSurveyInfo())
```
Comment (SteRiccio) [inline webapp/store/user/hooks.js:29]: very stupid thing, useAuthcanUseAnalysis should be useAuthCanUseAnalysis, just to be consistent with the other hooks

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: webapp/store/user/index.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ import {
   useAuthCanEditUser,
   useAuthCanInviteUser,
   useAuthCanViewOtherUsersEmail,
+  useAuthcanUseAnalysis,
```
Comment (ramirobg94) [inline webapp/store/user/index.js:13]:
```suggestion
  useAuthCanUseAnalysis,
```

File: webapp/store/user/index.js [hunk 3]
```diff
@@ -26,6 +27,7 @@ export {
...
```
Comment (ramirobg94) [inline webapp/store/user/index.js:30]:
```suggestion
  useAuthCanUseAnalysis,
```

File: webapp/store/user/hooks.js (hunk 1 above)
Comment (ramirobg94) [inline webapp/store/user/hooks.js:29]:
```suggestion
export const useAuthCanUseAnalysis = () => Authorizer.canAnalyzeRecords(useUser(), useSurveyInfo())
```

File: webapp/components/survey/SurveyDefsLoader/SurveyDefsLoader.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ import * as Survey from '@core/survey/survey'
 import { useI18n } from '@webapp/store/system'
 
 import { SurveyActions, useOnSurveyCycleUpdate, useSurveyDefsFetched, useSurveyInfo } from '@webapp/store/survey'
+import { useAuthcanUseAnalysis } from '@webapp/store/user'
```
Comment (ramirobg94) [inline webapp/components/survey/SurveyDefsLoader/SurveyDefsLoader.js:10]:
```suggestion
import { useAuthCanUseAnalysis } from '@webapp/store/user'
```

File: webapp/components/survey/SurveyDefsLoader/SurveyDefsLoader.js [hunk 5]
```diff
@@ -16,10 +17,11 @@ const SurveyDefsLoader = (props) => {
   const surveyInfo = useSurveyInfo()
   const ready = useSurveyDefsFetched(draft)
   const surveyUuid = Survey.getUuid(surveyInfo)
+  const includeAnalysis = useAuthcanUseAnalysis()
```
Comment (ramirobg94) [inline webapp/components/survey/SurveyDefsLoader/SurveyDefsLoader.js:20]:
```suggestion
  const includeAnalysis = useAuthCanUseAnalysis()
```

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**